packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -512,29 +512,17 @@ class _TreemapState extends State<Treemap> {
     );
   }
 
-  Container buildBreadcrumbsNavigator() {
+  Widget buildBreadcrumbNavigator() {
     final pathFromRoot = widget.rootNode.pathFromRoot();
-    return Container(
-      height: Treemap.treeMapHeaderHeight,
-      child: ListView.separated(
-        shrinkWrap: true,
-        scrollDirection: Axis.horizontal,
-        separatorBuilder: (context, index) {
-          return const Text(' > ');
-        },
-        itemCount: pathFromRoot.length,
-        itemBuilder: (BuildContext context, int index) {
-          return buildSelectable(
-            child: Text(
-              index < pathFromRoot.length - 1
-                  ? pathFromRoot[index].name
-                  : pathFromRoot[index].displayText(),
-            ),
-            newRoot: pathFromRoot[index],
-          );
-        },
-      ),
-    );
+    final breadcrumbs = List<Breadcrumb>.generate(pathFromRoot.length, (index) {
```

Comment:
nit: there probably won't be enough breadcrumbs to matter but it could be nice to continue to use itemBuilder like the code was previously so you don't build the breadcrumb widgets until they are needed.  you're already using `List.generate` so the api shape seems about identical. I would just have the Breadcrumb navigator take an arbitrary widget as a child so you can specify the ListView here like it was done before.

Reply:
Switched to a builder constructor so that we still get the benefit of ListView.builder but the consumer does not have to be responsible for the List widget code.